Marlin/src/lcd/extui/lib/mks_ui/draw_more.cpp

```diff
@@ -0,0 +1,267 @@
+#define ID_CUSTOM_4   4
+#define ID_CUSTOM_5   5
+#define ID_CUSTOM_6   6
+#define ID_CUSTOM_7   7
+#define ID_M_RETURN   8
+
+static void event_handler(lv_obj_t * obj, lv_event_t event) {
+  switch (obj->mks_obj_id) {
+    case ID_CUSTOM_1:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_1_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_1));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_2:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_2_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_2));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_3:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_3_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_3));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_4:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_4_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_4));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_5:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_5_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_5));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_6:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_6_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_6));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_7:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_7_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_7));
+        #endif
+      }
+      break;
+    case ID_M_RETURN:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        lv_clear_more();
+        lv_draw_tool();
+      }
+      break;
+  }
+}
+
+void lv_draw_more(void) {
+  #if ENABLED(USER_CMD_1_ENABLE)
+    lv_obj_t *buttonCustom1;
+  #endif
+  #if ENABLED(USER_CMD_2_ENABLE)
+    lv_obj_t *buttonCustom2;
+  #endif
+  #if ENABLED(USER_CMD_3_ENABLE)
+    lv_obj_t *buttonCustom3;
+  #endif
+  #if ENABLED(USER_CMD_4_ENABLE)
+    lv_obj_t *buttonCustom4;
+  #endif
+  #if ENABLED(USER_CMD_5_ENABLE)
+    lv_obj_t *buttonCustom5;
+  #endif
+  #if ENABLED(USER_CMD_6_ENABLE)
+    lv_obj_t *buttonCustom6;
+  #endif
+  #if ENABLED(USER_CMD_7_ENABLE)
+    lv_obj_t *buttonCustom7;
+  #endif
+  scr = lv_screen_create(MORE_UI);
+
+  /*Create an Image button and label*/
+
+  const bool enc_ena = TERN0(HAS_ROTARY_ENCODER, gCfgItems.encoder_enable);
+
+  #if ENABLED(USER_CMD_1_ENABLE)
+    buttonCustom1 = lv_imgbtn_create(scr, "F:/bmp_custom1.bin", INTERVAL_V, titleHeight, event_handler, ID_CUSTOM_1);
```

Comment:
Is `F:/` a path used on the display? It seems strange to see it here.

Reply:
It's a virtual drive from LVGL. It's used to abstract the load logic. Here, "F:/" is to load data from SPI Flash, and "S:" is to load data from SD.
https://docs.lvgl.io/latest/en/html/overview/file-system.html#add-a-driver\